python/python/bystro/supervised_ppca/ppca_adversarial.py

```diff
@@ -11,13 +11,18 @@
 of the adversarial component and the conditioning of inverses, 
 respectively, enhancing the robustness and stability of the model.
 """
-
+from typing import Union
```

Comment:
we don't really want/need to import union; the modern way to do this is to use a pipe character `int | None`